tool/get-dart/archive/compile.sh

```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")"
```

Comment:
Can you add a short comment above this explaining what the line does?

Reply:
```suggestion
# Enter the tool's directory as the working directory to avoid path conflicts under different environments.
cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")"
```